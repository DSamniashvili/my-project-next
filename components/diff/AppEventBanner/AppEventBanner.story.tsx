import {Meta, Story} from '@storybook/react';
import AppEventBanner from './AppEventBanner';

export default {
   title: 'components / diff / AppEventBanner',
   component: AppEventBanner,
} as Meta<typeof AppEventBanner>;

const Template: Story<typeof AppEventBanner> = (args) => <AppEventBanner />;

export const Default: Story<typeof AppEventBanner> = Template.bind({});

Default.args = {};
