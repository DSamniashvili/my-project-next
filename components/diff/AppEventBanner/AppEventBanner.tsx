'use client';
import React, {FC} from 'react';
import {
   SEventBannerCloseButton,
   SEventBannerContent,
   SEventBannerContentContainer,
   SEventBannerContentDesctiption,
   SEventBannerContentTitle,
   SEventBannerWrapper,
   SAppEventBannerContainer,
} from './SAppEventBanner.styled';
import {EventBannerProps} from './AppEventBanner.types';
import AppLinkButton from '@components/shared/AppLinkButton/AppLinkButton';
import {CloseButton} from '@assets/icons/CloseButton';

const AppEventBanner: FC<EventBannerProps> = ({
   title,
   description,
   linkText,
   url = '#',
}) => {
   // TODO - temporarily handling event banner in this way. Will be updated when all routes are ready!
   const handleCloseBanner = () => {
      window.location.reload();
   };

   return (
      <SEventBannerWrapper>
         <SAppEventBannerContainer>
            <SEventBannerContentContainer>
               <SEventBannerContent>
                  <SEventBannerContentTitle>{title}</SEventBannerContentTitle>
                  <SEventBannerContentDesctiption>
                     {description}
                  </SEventBannerContentDesctiption>
                  <AppLinkButton href={url} underline text={linkText} />
               </SEventBannerContent>
               <SEventBannerCloseButton onClick={handleCloseBanner}>
                  <CloseButton />
               </SEventBannerCloseButton>
            </SEventBannerContentContainer>
         </SAppEventBannerContainer>
      </SEventBannerWrapper>
   );
};

export default AppEventBanner;
