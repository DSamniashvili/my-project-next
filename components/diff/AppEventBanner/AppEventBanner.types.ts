export interface EventBannerProps {
   title: string;
   description: string;
   linkText: string;
   url: string;
}
