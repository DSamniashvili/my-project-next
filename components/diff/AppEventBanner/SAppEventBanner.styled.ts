import styled from 'styled-components';
import tw from 'twin.macro';

export const SEventBannerWrapper = styled.div`
   ${tw`bg-app-color-secondary flex justify-center items-center w-full`}
`;

export const SAppEventBannerContainer = styled.div`
   ${tw`bg-app-color-secondary py-app-padding-24 px-app-padding-100 flex justify-center items-center app-max-width-small:px-app-padding-24 app-max-width-small:py-app-padding-16  w-full`}
`;

export const SEventBannerContentContainer = styled.div`
   ${tw`flex justify-between items-center w-full`}
`;

export const SEventBannerContent = styled.div`
   ${tw`flex gap-5 items-center app-max-width-small:gap-1 app-max-width-extra-small:[width: 300px]`}
   a {
      p {
         ${tw`flex-none text-app-font-14 app-max-width-small:text-app-font-10`}
      }
   }
`;

export const SEventBannerContentTitle = styled.h3`
   ${tw`text-app-font-16 text-app-color-white uppercase font-bold flex-none app-max-width-extra-small:text-app-font-10 app-max-width-small:text-app-font-10`}
`;

export const SEventBannerContentDesctiption = styled.p`
   ${tw`text-app-font-16 text-app-color-white-09 m-0 flex-none`}

   @media screen and (max-width: 480px) {
      ${tw`text-app-font-10 inline-block whitespace-nowrap overflow-hidden text-ellipsis`}
      width: 30%;
   }
   @media screen and (max-width: 1024px) and (min-width: 481px) {
      ${tw`text-app-font-12`}
   }
`;

export const SEventBannerCloseButton = styled.button`
   svg {
      ${tw`app-max-width-small:w-2.5 app-max-width-small:h-2.5`}
      path {
         ${tw`fill-app-color-white`}
      }
   }
`;

SAppEventBannerContainer.defaultProps = {
   className: 'container',
};
