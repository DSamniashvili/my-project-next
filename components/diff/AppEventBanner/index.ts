import AppEventBanner from './AppEventBanner';
export {AppEventBanner as default};
