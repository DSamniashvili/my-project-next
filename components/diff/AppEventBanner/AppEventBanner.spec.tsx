import {render, screen} from '@testing-library/react';
import AppEventBanner from './AppEventBanner';

describe('[Component] AppEventBanner', () => {
   it('should render component', () => {
      render(<AppEventBanner />);

      expect(screen.getByText('test-title')).toBeInTheDocument();
   });

   it('should render correct title', () => {
      render(<AppEventBanner />);

      expect(screen.getByText('test-title')).toBeInTheDocument();
   });

   it('should render correct description', () => {
      render(<AppEventBanner />);

      expect(screen.getByText('test-description')).toBeInTheDocument();
   });

   it('should render correct linkText', () => {
      render(<AppEventBanner />);

      expect(screen.getByText('test-linkText')).toBeInTheDocument();
   });
});
