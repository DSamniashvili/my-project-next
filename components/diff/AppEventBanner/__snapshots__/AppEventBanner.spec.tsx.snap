// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] AppEventBanner should render component 1`] = `
<div>
  <div
    class="SAppEventBannerstyled__SEventBannerWrapper-sc-1czc15b-0 effGGI"
  >
    <div
      class="SAppEventBannerstyled__SAppEventBannerContainer-sc-1czc15b-1 cXnYOT container"
    >
      <div
        class="SAppEventBannerstyled__SEventBannerContentContainer-sc-1czc15b-2 hbvGET"
      >
        <div
          class="SAppEventBannerstyled__SEventBannerContent-sc-1czc15b-3 hDeden"
        >
          <button
            style="color: rgb(255, 255, 255);"
          >
            light
          </button>
          <h3
            class="SAppEventBannerstyled__SEventBannerContentTitle-sc-1czc15b-4 gaRFeE"
          >
            UPCOMING WEBINAR:
          </h3>
          <p
            class="SAppEventBannerstyled__SEventBannerContentDesctiption-sc-1czc15b-5 jRLQLC"
          >
            Market rate and development asia-to-europe sea freight
          </p>
          <p
            class="SAppLinkButtonstyled__SAppLinkButtonText-sc-9xu9ej-1 bULUXf"
          >
            Learn more
          </p>
        </div>
        <button
          class="SAppEventBannerstyled__SEventBannerCloseButton-sc-1czc15b-6 bwFXUv"
        >
          <title
            id="Close-Button"
          >
            Close button
          </title>
          <svg
            aria-labelledby="Close-Button"
            fill="none"
            height="15"
            role="img"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0.566406 14.501C0.712891 14.6475 0.882812 14.7441 1.07617 14.791C1.27539 14.8379 1.47168 14.8379 1.66504 14.791C1.8584 14.7441 2.02539 14.6504 2.16602 14.5098L7.4043 9.27148L12.6426 14.5098C12.7832 14.6504 12.9502 14.7441 13.1436 14.791C13.3369 14.8379 13.5303 14.8379 13.7236 14.791C13.917 14.7441 14.0869 14.6475 14.2334 14.501C14.3799 14.3545 14.4766 14.1846 14.5234 13.9912C14.5703 13.7979 14.5703 13.6045 14.5234 13.4111C14.4766 13.2178 14.3828 13.0508 14.2422 12.9102L9.0127 7.66309L14.2422 2.4248C14.3828 2.28418 14.4766 2.11719 14.5234 1.92383C14.5762 1.73047 14.5762 1.53711 14.5234 1.34375C14.4766 1.15039 14.3799 0.983398 14.2334 0.842773C14.0869 0.69043 13.917 0.59082 13.7236 0.543945C13.5303 0.49707 13.3369 0.49707 13.1436 0.543945C12.9502 0.59082 12.7832 0.68457 12.6426 0.825195L7.4043 6.06348L2.16602 0.825195C2.02539 0.68457 1.85547 0.59082 1.65625 0.543945C1.46289 0.49707 1.2666 0.49707 1.06738 0.543945C0.874023 0.59082 0.707031 0.6875 0.566406 0.833984C0.419922 0.980469 0.323242 1.15039 0.276367 1.34375C0.235352 1.53711 0.235352 1.7334 0.276367 1.93262C0.323242 2.12598 0.416992 2.29004 0.557617 2.4248L5.7959 7.66309L0.557617 12.9102C0.416992 13.0508 0.323242 13.2178 0.276367 13.4111C0.229492 13.6045 0.229492 13.7979 0.276367 13.9912C0.323242 14.1846 0.419922 14.3545 0.566406 14.501Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
